c7n/resources/secretsmanager.py

```diff
@@ -137,3 +141,90 @@ class MarkSecretForOp(TagDelayedAction):
                   op: tag
                   days: 1
     """
+
+
+@SecretsManager.action_registry.register('delete')
+class DeleteSecretsManager(BaseAction):
+    """Delete a secret and all of its versions.
+    The default recovery window is 30 days
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-unencrypted-secrets
+                resource: aws.secrets-manager
+                filters:
+                  - type: value
+                    key: KmsKeyId
+                    value: absent
+                actions:
+                  - delete
+    """
+
+    schema = type_schema('delete')
```

Comment:
should allow the ability to set the recovery window, default to 30 days

Reply:
also note in the boto3 docs:

> You can't delete a primary secret that is replicated to other Regions. You must first delete the replicas using RemoveRegionsFromReplication, and then delete the primary secret. When you delete a replica, it is deleted immediately.

we probably also want to add a restore-secret action as well just in case

Reply:
replication regions can be found in the resource dict under `ReplicationStatus[].Region`

Reply:
Sounds good, thanks for the feedback

Reply:
@thisisshi `restore-secret` action might not be possible because the secrets becomes invisible to the list/describe secret api call once they are marked for deletion. I went ahead and added an option to specify RecoveryWindow in days. Their apis don't let you check the recovery window parameter outside of the api call response.